### PR TITLE
Add additional Application Provider onboarding validation

### PR DESCRIPTION
**Pull request by kaloyanspiridonov**
**Description**

Changes proposed in this pull request:
- Drop db unique `name` constraint for app templates table
- When creating/updating an app template - add backend constraint which checks if the combination `name` + `region` is unique
- when creating app template:
- if `applicationType` label is missing in the applicationInput object - add it manually - "<app_template_name> (label[region])" or just "name" (if region label is not provided)
- if `applicationType` label is present in the applicationInput object - ensure that it fits the schema "<app_template_name> (label[region])"
- adapt registerApplicationFromTemplate to list all templates by name and take the one that is more local (the one labelled by the subaccount of the certificate making the request, if such exists)

**Pull Request status**
- [X] Implementation
- [X] Unit tests
- [X] Integration tests
- [X] `chart/compass/values.yaml` is updated 
- [X] Mocks are regenerated, using the automated script

### CODE REVIEW

File: components/director/internal/domain/apptemplate/resolver.go [hunk 1]
```diff
@@ -317,8 +326,8 @@ func (r *Resolver) RegisterApplicationFromTemplate(ctx context.Context, in graph
 	log.C(ctx).Infof("Registering an Application from Application Template with name %s", in.TemplateName)
 	convertedIn := r.appTemplateConverter.ApplicationFromTemplateInputFromGraphQL(in)
 
-	log.C(ctx).Debugf("Extracting Application Template with name %s from GraphQL input", in.TemplateName)
-	appTemplate, err := r.appTemplateSvc.GetByName(ctx, convertedIn.TemplateName)
+	log.C(ctx).Debugf("Extracting Application Template with name %q and subaccount %q from GraphQL input", in.TemplateName, consumerInfo.ConsumerID)
+	appTemplate, err := r.appTemplateSvc.GetByNameAndSubaccount(ctx, convertedIn.TemplateName, consumerInfo.ConsumerID)
```
[the remark this answers is not in the document]
Comment (NickyMateev) [reply components/director/internal/domain/apptemplate/resolver.go:330]: Just checked the implementation of the `GetByNameAndSubaccount` function and it seems like it has this in mind

File: components/director/internal/domain/apptemplate/service.go [hunk 2]
```diff
@@ -243,3 +350,91 @@ func (s *service) PrepareApplicationCreateInputJSON(appTemplate *model.Applicati
 	}
 	return appCreateInputJSON, nil
 }
+
+func (s *service) exists(ctx context.Context, inputName string, inputRegion interface{}) (bool, error) {
```
Comment (NickyMateev) [inline components/director/internal/domain/apptemplate/service.go:354]: The implementation of this function is identical to the GetByNameAndRegion function?

Comment (kaloyanspiridonov) [reply components/director/internal/domain/apptemplate/service.go]: yep. I had a note about that, but somehow i've missed it.

File: chart/compass/values.yaml [hunk 3]
```diff
@@ -637,6 +642,7 @@ global:
     systemsAPIFilterCriteria: ""
     # systemToTemplateMappings - how to map system properties to an existing application template
     systemToTemplateMappings: '{}'
+    templateRegion: "eu-1"
```
Comment (NickyMateev) [inline chart/compass/values.yaml:645]: Let's discuss this workaround again

Comment (kaloyanspiridonov) [reply chart/compass/values.yaml:645]: after an offline discussion we've decided to implement app template creation on system-fetcher start as next step

File: components/director/cmd/ns-adapter/main.go [hunk 4]
```diff
@@ -249,7 +249,7 @@ func registerAppTemplate(ctx context.Context, transact persistence.Transactioner
 		AccessLevel: model.GlobalApplicationTemplateAccessLevel,
 	}
 
-	_, err = appTemplateSvc.GetByName(ctxWithTx, appTemplateName)
+	_, err = appTemplateSvc.GetByNameAndRegion(ctxWithTx, appTemplateName, nil)
```
[the remark this answers is not in the document]
Comment (kaloyanspiridonov) [reply components/director/cmd/ns-adapter/main.go:252]: after an offline discussion we've decided to leave this method as it is.

**Review by NickyMateev — APPROVED** [review]

**Comment by NickyMateev** [comment]
lgtm

**Review by NickyMateev — APPROVED** [review]

**Pull request merged**